add-enc-exchange-set/src/main/java/com/esri/arcgismaps/sample/addencexchangeset/components/MapViewModel.kt

```diff
@@ -0,0 +1,112 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.addencexchangeset.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.geometry.Envelope
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.hydrography.EncCell
+import com.arcgismaps.hydrography.EncEnvironmentSettings
+import com.arcgismaps.hydrography.EncExchangeSet
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.EncLayer
+import com.esri.arcgismaps.sample.addencexchangeset.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+import java.io.File
+
+class MapViewModel(application: Application) : AndroidViewModel(application) {
+    private val provisionPath: String by lazy {
+        application.getExternalFilesDir(null)?.path.toString() + File.separator + application.getString(
+            R.string.app_name
+        )
+    }
+
+    // Paths to ENC data and hydrology resources
+    private val encResourcesPath = provisionPath + application.getString(R.string.enc_res_dir)
+    private val encDataPath = provisionPath + application.getString(R.string.enc_data_dir)
+
+    // Create an ENC exchange set from the local ENC data
+    private val encExchangeSet = EncExchangeSet(listOf(encDataPath))
+    private val encEnvironmentSettings: EncEnvironmentSettings = EncEnvironmentSettings
+
+    // Create a map with the oceans basemap style
+    val arcGISMap by mutableStateOf(ArcGISMap(BasemapStyle.ArcGISOceans))
```

Comment:
if you use a delegated property with mutable state, it should be `var`, otherwise if it's not going to change you can just expose the value directly